fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanSchedularServiceImpl.java

```diff
@@ -295,4 +311,48 @@ private void checkCompletion(List<Future<Void>> responses) {
             log.error("Execution exception while posting IR entries", e2);
         }
     }
+
+    @Override
+    @CronTarget(jobName = JobName.POST_LOAN_REPAYMENT_REMINDER)
+    public void postLoanRepaymentReminder() {
+        final AppUser currentUser = getAppUserIfPresent();
+        final List<LoanRepaymentReminderSettingsData> settingsData = loanRepaymentReminderSettingsRepository
+                .findLoanRepaymentReminderSettings();
+
+        if (!CollectionUtils.isEmpty(settingsData)) {
+            for (LoanRepaymentReminderSettingsData data : settingsData) {
+
+                final List<LoanRepaymentReminderData> repaymentReminders = loanReadPlatformService
+                        .findLoanRepaymentReminderData(data.getDays());
```

Comment:
@hero78 wouldn't some sort of pagination be important here?

Reply:
Yes Pagination and multi-Threading is the plan. I need MVP then come back to this